test/functional/test_solution.py

```diff
@@ -0,0 +1,439 @@
+# Copyright 2106 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import decorator
+import paramiko
+import pytest
+import time
+import sys
+
+from f5.bigip import BigIP
+from pprint import pprint as pp
+from neutronclient.v2_0 import client
+from f5_os_test.polling_clients import NeutronClientPollingManager
+
+
+lbaas_version = 1
+
+
+def log_test_call(func):
+    def wrapper(func, *args, **kwargs):
+        print("\nRunning %s" % func.func_name)
+        return func(*args, **kwargs)
+    return decorator.decorator(wrapper, func)
+
+
+class ExecTestEnv(object):
+    def __init__(self, symbols):
+        self.symbols = symbols.copy()
+        new_symbols = {
+            # leave the following settings alone
+            'bigip_username':           'admin',
+            'bigip_password':           'admin',
+            'lb_provider':              ('F5NetworksTest' if lbaas_version == 2
+                                         else 'f5'),
+            'admin_name':               'admin',
+            'admin_username':           'admin',
+            'admin_password':           'changeme',
+            'tenant_name':              'testlab',
+            'tenant_username':          'testlab',
+            'tenant_password':          'changeme',
+            'client_subnet':            'client-v4-subnet',
+            'guest_username':           'centos',
+            'guest_password':           'changeme',
+            'server_http_port':         '8080',
+            'server_client_ip':         '10.2.2.3'
+        }
+        self.symbols.update(new_symbols)
+
+
+def nclientmanager(symbols):
+    nclient_config = {
+        'username': symbols['tenant_username'],
+        'password': symbols['tenant_password'],
+        'tenant_name': symbols['tenant_name'],
+        'auth_url': symbols['openstack_auth_url']
+    }
+
+    neutronclient = client.Client(**nclient_config)
+    return NeutronClientPollingManager(neutronclient)
+
+
+class LBaaSv1(object):
```

Comment:
Why is the LBaaSv1 object in the LBaaSv2 repo?  Is there a plan to migrate this to the LBaaSv1 repo? I suppose this implies that some of the code it uses is also migrated to a common library that all OpenStack tests would import?